connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -391,7 +391,15 @@ public List<Map<String, String>> connectorTaskConfigs(String connName, Connector
             Connector connector = workerConnector.connector();
```

Comment:
I didn't realize that this was the only place where Connector escapes the WorkerConnector, and a different thread interacts with the Connector object.

I know the taskConfigs method is typically an instantaneous method, but maybe it would make sense for this to eventually move to the WorkerConnector thread instead of the herder tick thread.

Reply:
It definitely would make sense to do this, especially since it would allow the herder to be resilient against blocked invocations of `taskConfigs`.